### PR TITLE
GraphQL support

### PR DESCRIPTION
**Pull request by excitement-engineer**
This PR adds [GraphQL](https://graphql.org/) support to Ktor: `ktor-graphql`. 

The library exposes a `graphQL` route. This needs to be setup with a path and an instance of a GraphQL schema as exposed by [graphql-java](https://github.com/graphql-java/graphql-java).

```
val server = embeddedServer(Netty, port = 8080) {
        routing {
            graphQL("/graphql", schema) {
                config {
                    graphiql = true
                }
            }
        }
    }
server.start(wait = true)
```

I hope that this library will make it very easy to start using GraphQL in combination with kotlin (and ktor).

Full documentation can be found in the README in this PR.

I have also created an [example repo](https://github.com/excitement-engineer/ktor-graphql-example) that provides a full implementation of a simple graphql server.

I'm curious to hear what everyone thinks of this and would love to hear feedback:)

### CODE REVIEW

**Comment by orangy** [comment]
Thanks for your PR! We are currently very busy with Kotlin 1.3 and KotlinConf and will be able to review it afterward.

**Comment by excitement-engineer** [comment]
No problem! Good luck with all the conference and the new release. I am looking forward to watching the conference online:)

**Comment by excitement-engineer** [comment]
Any updates for this pr?

**Comment by NeroBlackstone** [comment]
I hope ktor can support GraphQL as soon as possible.

File: ktor-features/ktor-graphql/src/io/ktor/graphql/renderGraphiQL.kt [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    #graphiql {
+        height: 100vh;
+    }
+    </style>
+    <link href="//cdn.jsdelivr.net/npm/graphiql@${GRAPHIQL_VERSION}/graphiql.css" rel="stylesheet" />
+    <script src="//cdn.jsdelivr.net/es6-promise/4.0.5/es6-promise.auto.min.js"></script>
+    <script src="//cdn.jsdelivr.net/fetch/0.9.0/fetch.min.js"></script>
+    <script src="//cdn.jsdelivr.net/react/15.4.2/react.min.js"></script>
+    <script src="//cdn.jsdelivr.net/react/15.4.2/react-dom.min.js"></script>
+    <script src="//cdn.jsdelivr.net/npm/graphiql@${GRAPHIQL_VERSION}/graphiql.min.js"></script>
```
Comment (JLLeitschuh) [inline ktor-features/ktor-graphql/src/io/ktor/graphql/renderGraphiQL.kt]: Can you include the subresource integrity check hashes? Preferably something SHA-256 or higher.

https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity

**Comment by adamsz-lume** [comment]
Could I get an official position of ktor creators on the topic, please? What would be the strategy to bring graphql to ktor?

I have started an initiative to bring graphql to ktor using graphql-kotlin library in this PR. I want to make sure I am following the right path here.

I can see the PR mentioned in this issue is quite a chank of work and I am not sure what it was not reviewed and possibly merged in.

**Comment by kaushalyap** [comment]
@orangy Any plans to support GraphQL?

**Pull request closed without merging**